### PR TITLE
Stream Enrich: bump amazon-kinesis-client to 2.2.12 (Closes #260)

### LINKED ISSUE #260: Stream Enrich: bump amazon-kinesis-client to 2.2.12
**Project**: Stream Enrich
**Version**: 1.1.0
**Expected behavior**: Use enhanced fan out
**Actual behavior**: KCL v1 missing it.

We had this requirement to use the enhance fan out feature from Kinesis to make sure our throughput is consistent. However, KCL v1 is not supporting this feature and we have to upgrade the KCL to version 2. This could be similar topic to https://github.com/snowplow/snowplow-elasticsearch-loader/issues/151.

### PR DESCRIPTION
**Pull request by szareiangm**
Added the KCL v2.

Notes:
- I will have a PR shortly to update AWS SDK.
- The CI/CD fails due to unrelated errors. How else can we test it?

### CODE REVIEW

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala [hunk 1]
```diff
@@ -91,7 +91,7 @@ object MapTransformer {
   )(
     implicit m: Manifest[T]
   ): ValidatedNel[FailureDetails.EnrichmentFailure, T] = {
-    val newInst = m.runtimeClass.newInstance()
```
Comment (chuwy) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94]: I'd restrain from making changes in SCE in this very SE-specific ticket. Would be much better if it had its dedicated commit: https://github.com/snowplow/enrich/issues/252

**Comment by chuwy** [comment]
Added `cla:yes` label manually - will enable CLA bot in this repo soon.

**Comment by chuwy** [comment]
> The CI/CD fails due to unrelated errors. How else can we test it?

Let me try to handle it.

**Comment by chuwy** [comment]
One more thing, @szareiangm - I'd recommend you to branch off from `release/1.3.0` branch. It has very small semantic changes (mostly around build process), but huge patch because we updated scalafmt. It will be much easier to rebase it later.

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala (hunk 1 above)
Comment (szareiangm) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94]: Sure. It was failing my compilation due to this error: 
```
[error] /enrich/modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94:34: method newInstance in class Class is deprecated: see corresponding Javadoc for more information.
[error]     val newInst = m.runtimeClass.newInstance()
```

Comment (chuwy) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94]: No, it can be same PR. Just separate commit.

Just in case - it fails because you use JDK 9 or higher. It compiles just fine on JDK 8.

Comment (szareiangm) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala:94]: It was very painful to pull out the JDK 13/14 from my system. back to work now. I will submit the fix for that one too.

**Comment by szareiangm** [comment]
My next PR is based on this PR and ready here: https://github.com/globeandmail/enrich/pull/1

We will test them internally in a regression environment and update here.

**Comment by chuwy** [comment]
Scheduling for 1.4.0. Will merge this into `develop` branch as soon as 1.3.0 is out (should be early next week).

**Comment by chuwy** [comment]
Cherry-picked with rebasing into: https://github.com/snowplow/enrich/pull/288. It will go into 1.4.0
